### PR TITLE
perf: avoid force reflow in use-size

### PR DESCRIPTION
**Pull request by kongmoumou**
[[中文版模板 / Chinese template](https://github.com/alibaba/hooks/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [x] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 💡 Background and solution

目前的实现会导致每次 render 都强制重新 layout，读取 el.clientWidth 的逻辑应该放进 initializer function 里防止重复调用

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English |           |
| 🇨🇳 Chinese |           |

### ☑️ Self Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: packages/hooks/src/useSize/index.ts [hunk 1]
```diff
@@ -9,7 +9,7 @@ type Size = { width: number; height: number };
 function useSize(target: BasicTarget): Size | undefined {
   const el = getTargetElement(target);
   const [state, setState] = useRafState<Size | undefined>(
-    el ? { width: el.clientWidth, height: el.clientHeight } : undefined,
+    () => el ? { width: el.clientWidth, height: el.clientHeight } : undefined,
   );
```
Comment (liuyib) [inline packages/hooks/src/useSize/index.ts:16]: How about putting `const el = getTargetElement(target);` into `useRafState(`'s callback, just like:

```ts
const [state, setState] = useRafState<Size | undefined>(() => {
  const el = getTargetElement(target);

  return el ? { width: el.clientWidth, height: el.clientHeight } : undefined;
});
```

So `const el = getTargetElement(target);` will be executed only on the initial render.

**Comment by liuyib** [comment]
@li-jia-nan  It is a better suggestion. If the initial state is the result of an expensive computation, `react` recommend to provide a function instead. see: https://reactjs.org/docs/hooks-reference.html#lazy-initial-state

image

File: packages/hooks/src/useSize/index.ts (hunk 1 above)
Comment (kongmoumou) [reply packages/hooks/src/useSize/index.ts:16]: Yep, it seems that the size only need to be initialized when target was already exist.

**Comment by kongmoumou** [comment]
@li-jia-nan could u make a review 😂

**Review by liuyib — APPROVED** [review]

**Comment by liuyib** [comment]
@crazylxr  ready to merge

**Review by crazylxr — APPROVED** [review]

**Pull request merged**